app/pocket.go

```diff
@@ -87,12 +94,12 @@ func (app *PocketCoreApp) InitChainerWithGenesis(ctx sdk.Ctx, req abci.RequestIn
 
 // setups all of the begin blockers for each module
 func (app *PocketCoreApp) BeginBlocker(ctx sdk.Ctx, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
-	return app.mm.BeginBlock(ctx, req)
+	return app.mm.BeginBlock(ctx.WithCache(app.GlobalCacheStore), req)
```

Comment:
I'm worried about the ctx coming from other places (Queries and Transactions)

Setting .WithCache every-time seems error prone... Maybe we can add the cache anytime you create NewContext()?

Reply:
resolved by pushing the cache storage to the context level in its entirety